### PR TITLE
Save Reward Settings In Metadata

### PR DESCRIPTION
**Pull request by Kennybll**
Fixes Save reward settings (50%/100% and upvote..) in metadata.

Changes:
- Saves previous post's reward settings under SteemConnect.
- Then fetched it when creating a new post/draft.

### CODE REVIEW

File: src/post/Write/editorActions.js [hunk 1]
```diff
@@ -31,6 +31,11 @@ export const addEditedPost = createAction(ADD_EDITED_POST);
 export const DELETE_EDITED_POST = '@editor/DELETE_EDITED_POST';
 export const deleteEditedPost = createAction(DELETE_EDITED_POST);
 
+export const UPDATE_LAST_SETTINGS = '@editor/UPDATE_LAST_SETTINGS';
```
Comment (jm90m) [inline src/post/Write/editorActions.js]: you can use `createAsyncActionType` from `helpers/stateHelpers'

Comment (jm90m) [reply src/post/Write/editorActions.js]: and then you won't need the unnecessary START, SUCCESS, ERROR actions in here.

**Comment by jm90m** [comment]
Is there an issue tied to this?

**Comment by Kennybll** [comment]
No. Thought it was added under Todos in the workflow.

**Comment by Kennybll** [comment]
@jm90m do you think we should use createAsyncActionType on all promise based redux actions or no? I mean app wide? I can do that with another pull request. What do you think @bonustrack

**Comment by bonustrack** [comment]
What your thought @Sekhmet ?

**Comment by Kennybll** [comment]
It would clean up the code, and there would be less code to look through...

**Comment by jm90m** [comment]
yeah me @Sekhmet talked about using it app wide, but it can be done in another PR. It's not that important right now, their are more pressing issues to work on.

**Comment by Kennybll** [comment]
So it it good?

**Comment by bonustrack** [comment]
@jm90m I think it's something @Kennybll can do. I told him that we need these kind of changes to reduce our code complexity.

**Comment by jm90m** [comment]
yeah then its good to do it in different PR, thanks!

**Comment by bonustrack** [comment]
I've published a post with reward declined and i've uncheck "Like this post". When i try to publish a new post, the reward select is properly set to "Declined" but the "Like this post" checkbox is checked, that should not be the case:
image

**Comment by Kennybll** [comment]
Question, If I were refactoring to use the createAsyncActionType would I put a new line between something like:
```export const GET_COMMENTS = createAsyncActionType('GET_COMMENTS');

export const SEND_COMMENT = createAsyncActionType('SEND_COMMENT');```

**Comment by Kennybll** [comment]
export const GET_COMMENTS = createAsyncActionType('GET_COMMENTS');

export const SEND_COMMENT = createAsyncActionType('SEND_COMMENT');

**Comment by bonustrack** [comment]
It's not necessary to add line between these 2 @Kennybll

**Comment by Kennybll** [comment]
okay

**Comment by Kennybll** [comment]
@bonustrack Did it work?

**Comment by Kennybll** [comment]
Wait, should I push to the branch 'master'? @bonustrack

**Comment by Sekhmet** [comment]
You have to push to `Kennybll:rewards-settings`

**Comment by Kennybll** [comment]
I meant merge into master @Sekhmet

File: src/helpers/metadata.js [hunk 2]
```diff
@@ -19,6 +19,20 @@ export const saveSettingsMetadata = settings =>
     )
     .then(resp => resp.user_metadata.settings);
 
+export const saveUpvoteRewardsSettingsMetadata = (upvote, rewards) =>
```
Comment (Sekhmet) [inline src/helpers/metadata.js]: Can't we use `saveSettingsMetadata` for this?

File: src/post/Write/Write.js [hunk 3]
```diff
@@ -53,7 +57,9 @@ class Write extends React.Component {
     draftPosts: PropTypes.shape().isRequired,
     loading: PropTypes.bool.isRequired,
     saving: PropTypes.bool,
+    upvoteSetting: PropTypes.bool,
     draftId: PropTypes.string,
+    rewardSetting: PropTypes.string,
```
Comment (Sekhmet) [inline src/post/Write/Write.js]: Can we use the same order of propTypes and defaultProps?

**Comment by Kennybll** [comment]
There @Sekhmet .

File: src/post/Write/editorReducer.js [hunk 4]
```diff
@@ -42,60 +47,77 @@ const editor = (state = defaultState, action) => {
         success: false,
         loadingImg: false,
       };
-    case editorActions.CREATE_POST_START:
+    case editorActions.CREATE_POST.START:
```
Comment (Sekhmet) [inline src/post/Write/editorReducer.js]: This and other actions below it (except `UPDATE_LAST_SETTINGS`) won't work now.
You forgot to update this after removing `createAsyncActionType`.

File: src/post/Write/editorReducer.js [hunk 5]
```diff
@@ -10,6 +10,8 @@ const defaultState = {
   draftPosts: {},
   pendingDrafts: [],
   editedPosts: [],
+  upvoteSetting: true,
```
Comment (Sekhmet) [inline src/post/Write/editorReducer.js]: I think that those should belong to settings, not the editor.

**Comment by jm90m** [comment]
Branch is conflicting with lots of files, please fix this and then we can review when we have time. Thanks again!

**Comment by Kennybll** [comment]
I'm going to close this and open another. This is made to get merged into the new-design branch.

**Pull request closed without merging**